tech-debt: Replace PATTERNS.moduleType regex in config.ts with bridge roxenDetect()

## Description
Replace the PATTERNS.moduleType regex in config.ts with a call to bridge.roxenDetect() which returns module_type information.

## Problem
config.ts line 58 uses regex /constant\s+(?:int\s+)?module_type\s*=\s*(MODULE_\w+)/i to extract the module_type from source code. This fails on multi-line declarations, conditional compilation, and comments. The bridge's roxenDetect() already returns this info correctly via Parser.Pike.

## Expected Behavior
parseRoxenConfig() should use bridge.roxenDetect().module_type to get the module type constant, falling back to regex only when bridge data is unavailable.

## Suggested Approach
1. Read packages/pike-lsp-server/src/features/roxen/config.ts (269 lines)
2. Read packages/pike-bridge/src/bridge.ts roxenDetect() method (around line 2118)
3. Modify parseRoxenConfig() to accept optional bridge data parameter
4. Replace PATTERNS.moduleType usage at line 99 with bridge data when available
5. Keep regex as fallback for when bridge data is unavailable
6. Run: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json
7. Run: bun test packages/pike-lsp-server/src/tests/

## Affected Files
packages/pike-lsp-server/src/features/roxen/config.ts

## Acceptance
- [ ] Code change implemented
- [ ] TypeScript compiles: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json
- [ ] Smoke tests pass

**Parent issue:** #1338

**Compile failed:**

shell: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1
  stdout: packages/pike-lsp-server/src/core/index.ts(9,50): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(10,33): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(11,34): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(8,33): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(166,20): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(206,21): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(212,26): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/code-actions.ts(25,24): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/inde

**Compile failed:**

shell: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1
  stdout: packages/pike-lsp-server/src/core/index.ts(9,50): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(10,33): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(11,34): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/code-actions.ts(25,24): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/code-lens.ts(14,24): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/color-presentation.ts(21,24): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/document-links.ts(12,24): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/packages/core/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/folding.ts(17,24): error TS6305: Output file '/home/node/pike-lsp-slot1/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot1/

**Compile failed:**

shell: bunx tsc --noEmit -p packages/pike-lsp-server/tsconfig.json 2>&1
  stdout: packages/pike-lsp-server/src/core/index.ts(9,50): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(10,33): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/index.ts(11,34): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(8,33): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(166,20): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(206,21): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/core/types.ts(212,26): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/pike-bridge/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/pike-bridge/src/index.ts'.
packages/pike-lsp-server/src/features/advanced/code-actions.ts(25,24): error TS6305: Output file '/home/node/pike-lsp-slot0/packages/core/dist/src/index.d.ts' has not been built from source file '/home/node/pike-lsp-slot0/packages/core/src/inde

**coder report:**

Let me read the target files first.Now let me find the `RoxenModuleInfo` type to understand the return shape:Now let me read the detector.ts to understand how roxenDetect is currently used alongside config.ts:Now I understand the full picture. Let me check where `parseRoxenConfig` is actually called to understand how callers would pass roxen data:Now I have full context. The code has already been partially refactored (no `PATTERNS.moduleType` regex exists — it was replaced with symbol-based detection). The issue asks to also accept `RoxenModuleInfo` from `bridge.roxenDetect()` as a preferred source for module_type.

Let me check the types import:Now let me check where the types file is for the RoxenModuleInfo type re-export:Now I have full understanding. The plan:

1. Add `RoxenModuleInfo` import to config.ts
2. Extend `BridgeParseInput` with optional `roxenInfo?: RoxenModuleInfo`
3. In `parseRoxenConfig`, use `roxenInfo.module_type[0]` as primary source for `moduleType` and `roxenInfo.inherits` for `isInheritModule`, falling back to symbol-based detection, then string scanning

Let me also check how `roxenInfo` data maps:
- `isInheritModule`: `roxenInfo.is_roxen_module === 1` and/or `roxenInfo.inherits` contains `"module"` or `"roxen"`
- `moduleType`: `roxenInfo.module_type[0]` (first MODULE_* constant)

Now implement:Now update `parseRoxenConfig` to use `roxenInfo` as the primary source:I see a duplicate comment block. Let me fix that:Good. Now let's compile and test:Syntax error. Let me see what's at line 165:The `/**` opening was removed along with the duplicate. Let me fix:The errors are pre-existing (node type definitions missing, deprecated moduleResolution). Let me check if this was the state before my change:Same errors — no new errors from my change. The pre-existing issues are unrelated. Let me try with a stricter check to confirm no type errors from my code:Only one new error from my change:
- `packages/pike-lsp-server/src/features/roxen/config.ts(185,7): error TS7006: Parameter 'name' implicitly has an 'any' type.`

The `.some()` callback parameter needs a type. Let me fix it:Now build the dependent packages first, then typecheck:No errors in config.ts. The exit code 1 is from pre-existing errors in other files. Let me confirm no new errors by comparing against the total count:Clean — zero errors in config.ts. Now run the config tests:All 48 tests pass. Let me also verify the full test suite for the roxen feature:All 255 tests pass across the entire roxen test suite. Let me verify the final state of the changed file:Clean implementation. Let me verify line count stays under 500:381 lines, well under the 500-line limit.

**Summary of changes:**

One file changed: `packages/pike-lsp-server/src/features/roxen/config.ts`

1. **Import**: Consolidated `PikeSymbol` and `PikeToken` imports into a single line and added `RoxenModuleInfo` from `@pike-lsp/pike-bridge`.
2. **`BridgeParseInput` interface**: Added `roxenInfo?: RoxenModuleInfo` fiel

PR created: https://github.com/TheSmuks/pike-lsp/pull/1367

--- a/packages/pike-lsp-server/src/features/roxen/config.ts
+++ b/packages/pike-lsp-server/src/features/roxen/config.ts
@@ -10,8 +10,7 @@
 
 import type { CompletionItem, Diagnostic, Position } from 'vscode-languageserver';
 import { CompletionItemKind, DiagnosticSeverity } from 'vscode-languageserver/node.js';
-import type { PikeSymbol } from '@pike-lsp/pike-bridge';
-import type { PikeToken } from '@pike-lsp/pike-bridge';
+import type { PikeSymbol, PikeToken, RoxenModuleInfo } from '@pike-lsp/pike-bridge';
 import { TYPE_CONSTANTS, MODULE_CONSTANTS, VAR_FLAGS } from './constants.js';
 import {
   extractDefvarsFromTokens,
@@ -54,10 +53,16 @@ export interface ConfigError {
 
 /**
  * Optional inputs from Pike bridge for parser-backed analysis.
- * When provided, inherit and constant detection uses bridge parse results
- * instead of string scanning. Tokens enable defvar extraction via token walking.
+ *
+ * Priority order for module_type and inherit detection:
+ * 1. roxenInfo (bridge.roxenDetect()) — most authoritative, uses Parser.Pike
+ * 2. symbols (bridge parse) — parser-backed symbol table
+ * 3. String scanning fallback — last resort
+ *
+ * Tokens enable defvar extraction via token walking.
  */
 export interface BridgeParseInput {
+  roxenInfo?: RoxenModuleInfo;
   symbols?: PikeSymbol[];
   tokens?: PikeToken[];
 }
@@ -160,8 +165,10 @@ function detectModuleType(code: string, symbols?: PikeSymbol[]): string | null {
 /**
  * Parse Roxen configuration from code.
  *
- * When BridgeParseInput is provided with symbols/tokens from the Pike bridge,
- * uses parser-backed analysis (ADR-001). Otherwise falls back to string scanning.
+ * When BridgeParseInput is provided, uses bridge data in priority order:
+ * 1. roxenInfo (bridge.roxenDetect()) — authoritative, uses Parser.Pike
+ * 2. symbols/tokens — parser-backed symbol table and token walking
+ * 3. String scanning — fallback
  */
 export function parseRoxenConfig(code: string, bridgeInput?: BridgeParseInput): RoxenConfig {
   const result: RoxenConfig = {
@@ -171,8 +178,23 @@ export function parseRoxenConfig(code: string, bridgeInput?: BridgeParseInput): 
     errors: [],
   };
 
-  result.isInheritModule = detectInheritModule(code, bridgeInput?.symbols);
-  result.moduleType = detectModuleType(code, bridgeInput?.symbols);
+  // Priority 1: roxenInfo from bridge.roxenDetect()
+  if (bridgeInput?.roxenInfo && bridgeInput.roxenInfo.is_roxen_module === 1) {
+    const info = bridgeInput.roxenInfo;
+    const inheritsModuleOrRoxen = info.inherits.some(
+      (name: string) => name === 'module' || name === 'roxen'
+    );
+    result.isInheritModule = inheritsModuleOrRoxen;
+    if (info.module_type.length > 0) {
+      result.moduleType = info.module_type[0]!;
+    } else {
+      result.moduleType = detectModuleType(code, bridgeInput.symbols);
+    }
+  } else {
+    // Priority 2: symbols from bridge parse, 3: string scanning
+    result.isInheritModule = detectInheritModule(code, bridgeInput?.symbols);
+    result.moduleType = detectModuleType(code, bridgeInput?.symbols);
+  }
 
   // Extract defvars: prefer token-based extraction, fall back to code scanning
   if (bridgeInput?.tokens && bridgeInput.tokens.length > 0) {

